client/app/components/land-use-action.js

```diff
@@ -92,29 +96,29 @@ export default class LandUseActionComponent extends Component {
   }
 
   // when a user selects an option from the dropdown menu,
-  // the option is removed from the landUseActionOptions
+  // the option is filtered out of the availableLandUseActionOptions
   // (and therefore removed from the dropdown).
   // The selected option is added to the newLandUseActionSelections array
   // (and therefore displays in the list of selected actions).
   @action
   addNewLandUseActionSelection() {
-    // use unshiftObject to place new action at top of list
     const selection = this.selectedLandUseAction.field;
 
     if (selection) {
-      this.newLandUseActionSelections.unshiftObject(this.selectedLandUseAction);
+      this.newLandUseActionSelections.pushObject(this.selectedLandUseAction);
```

Comment:
Interesting, was initially confused what `pushObject` was because I couldn't find it in JS docs, before learning that it's an Ember array method.

Why do we need to switch to using `pushObject` from `unshiftObject`?

Reply:
Originally, when we only had one array that made up the selected actions list, we wanted new actions to show up at the top of the list -- so `unshiftObject` was used rather than `pushObject` (which adds to the end of an array). Now that we are combining two arrays to create that selected actions list, pushing the object to the front of the "new" actions array won't actually put it at the top of the actions list in the UI. Going to make a new issue for making sure most recently added actions show up at the top of the list.

Reply:
That's super helpful, thanks for clarifying!

Reply:
Yeah, this is a case where Ember's template observers need to be notified through this method.

It's akin to needing to call ".setState" in React.

I'm not sure if it's still true, though, but we use it out of habit. Would be worth researching.